common/core/devdoc/shared_access_signature.md

```diff
@@ -0,0 +1,74 @@
+SRS_NODE_COMMON_SAS_05_011: [The sr property shall have the value of resourceUri.]
+SRS_NODE_COMMON_SAS_05_012: [The sig property shall be the result of URL-encoding the value <signature>.]
+SRS_NODE_COMMON_SAS_05_013: [<signature> shall be an HMAC-SHA256 hash of the value <stringToSign>, which is then base64-encoded.]
+SRS_NODE_COMMON_SAS_05_014: [<stringToSign> shall be a concatenation of resourceUri + '\n' + expiry.]
+SRS_NODE_COMMON_SAS_05_015: [The se property shall have the value of expiry.]
+SRS_NODE_COMMON_SAS_05_016: [The skn property shall be the value <urlEncodedKeyName>.]
+SRS_NODE_COMMON_SAS_05_017: [<urlEncodedKeyName> shall be the URL-encoded value of keyName.]
+SRS_NODE_COMMON_SAS_05_018: [If the keyName argument to the create method was falsy, skn shall not be defined.]
+
+### parse(source, [requiredFields]) [static]
+The `parse` static method returns a new instance of the SharedAccessSignature object with properties corresponding to each 'name=value' field found in source.
+
+SRS_NODE_COMMON_SAS_05_001: [The input argument source shall be converted to string if necessary.]
+SRS_NODE_COMMON_SAS_05_002: [The parse method shall create a new instance of SharedAccessSignature.]
+SRS_NODE_COMMON_SAS_05_003: [It shall accept a string argument of the form 'name=value[&name=value…]' and for each name extracted it shall create a new property on the SharedAccessSignature object instance.]
+SRS_NODE_COMMON_SAS_05_004: [The value of the property shall be the value extracted from the source argument for the corresponding name.]
+SRS_NODE_COMMON_SAS_05_005: [The parse method shall throw FormatError if the shared access signature string does not start with 'SharedAccessSignature<space>'.]
+SRS_NODE_COMMON_SAS_05_006: [The parse method shall throw ArgumentError if any of fields in the requiredFields argument are not found in the source argument.]
+SRS_NODE_COMMON_SAS_05_007: [The generated SharedAccessSignature object shall be returned to the caller.]
+
+### toString()
+Creates a string representation of the SharedAccessSignature.
+
+SRS_NODE_COMMON_SAS_05_019: [The toString method shall return a shared-access signature token of the form: SharedAccessSignature sr=<resourceUri>&sig=<urlEncodedSignature>&se=<expiry>&skn=<urlEncodedKeyName>]
+SRS_NODE_COMMON_SAS_05_020: [The skn segment is not part of the returned string if the skn property is not defined.]
+
+### createWithSigningFunction(credentials: authorization.TransportConfig, expiry: number, signingFunction: Function, callback: (err: Error, sas?: SharedAccessSignature) => void) [static]
+Creates a `SharedAccessSignature` utilizing the supplied `signingFunction`.  If the signing operation is successful the results will be made available via the `callback`
+
+**SRS_NODE_COMMON_SAS_06_001: [** If `credentials`, `expiry`, `signingFunction`, or `callback` are falsy, `createWithSigningFunction` shall throw ReferenceException. **]**
```

Comment:
nit: ReferenceError. ReferenceException does not exist

Reply:
Derp.

---
In reply to: [165243853](https://github.com/Azure/azure-iot-sdk-node/pull/217#discussion_r165243853) [](ancestors = 165243853)